source/server-selection/server-selection.rst

```diff
@@ -341,6 +341,14 @@ smallestMaxStalenessSeconds
 A constant, 90 seconds. See "Smallest allowed value for maxStalenessSeconds"
 in the Max Staleness Spec.
 
+serverSelector
+~~~~~~~~~~~~~~
+
+Implementations MAY allow configuration of an optional, application-provided function
+that augments the server selection rules.  The function takes as a parameter a list
+of server descriptions representingthe suitable servers for the read or write operation,
```

Comment:
"representing the"

Reply:
I propose we also pass in the TopologyDescription or at least the TopologyType. This extra info is needed to implement some of our own read preferences, might be necessary for the application logic too.

Reply:
There's a lot of cruft in the topology description so I'd prefer just to add the topology type.  One thing to consider though: unlike the driver's generic selectors, the application author generally will know what topology type is being connected to rather than needing it to be discovered, so knowledge of the type could be backed into the function itself.

Reply:
That's fine. Either including the topology type, or not changing this at all, is ok with me.